servicetalk-http-utils/src/main/java/io/servicetalk/http/utils/TimeoutHttpRequesterFilter.java

```diff
@@ -174,10 +172,37 @@ public StreamingHttpConnectionFilter create(final FilterableStreamingHttpConnect
         return new StreamingHttpConnectionFilter(connection) {
             @Override
             public Single<StreamingHttpResponse> request(final StreamingHttpRequest request) {
-                HttpExecutionContext executionContext = connection.executionContext();
-                return TimeoutHttpRequesterFilter.this.withTimeout(request, r -> delegate().request(r),
-                        executionContext.executionStrategy().hasOffloads() ?
-                                executionContext.executor() : executionContext.ioExecutor());
+                final FilterableStreamingHttpConnection delegate = delegate();
+                return TimeoutHttpRequesterFilter.this.withTimeout(request,
+                        delegate.connectionContext().protocol().major() > 1 ? delegate::request :
+                                r -> delegate.request(r).liftSync(new BeforeFinallyHttpOperator(
+                                        new TerminalSignalConsumer() {
+                                            @Override
+                                            public void onComplete() {
+                                                // noop
+                                            }
+
+                                            @Override
+                                            public void onError(final Throwable throwable) {
+                                                // noop
+                                            }
+
+                                            @Override
+                                            public void cancel() {
+                                                // In alignment with cancellation processing in
```

Comment:
Consider moving the comments up to the conditional operator.

Reply:
Having the comment here helps to quickly find a similar callback inside `LoadBalancedStreamingHttpClient` that has more clarification around the same topic. Let me keep it here.